pkg/querier/remote_read.go

```diff
@@ -6,78 +6,159 @@
 package querier
 
 import (
+	"context"
+	"io"
 	"net/http"
 
 	"github.com/go-kit/log"
 	"github.com/go-kit/log/level"
+	"github.com/gogo/protobuf/proto"
+	"github.com/pkg/errors"
 	"github.com/prometheus/prometheus/storage"
+	prom_remote "github.com/prometheus/prometheus/storage/remote"
 
 	"github.com/grafana/mimir/pkg/ingester/client"
 	"github.com/grafana/mimir/pkg/mimirpb"
 	"github.com/grafana/mimir/pkg/util"
 	util_log "github.com/grafana/mimir/pkg/util/log"
 )
 
-// Queries are a set of matchers with time ranges - should not get into megabytes
-const maxRemoteReadQuerySize = 1024 * 1024
+const (
+	// Queries are a set of matchers with time ranges - should not get into megabytes
+	maxRemoteReadQuerySize = 1024 * 1024
+)
 
 // RemoteReadHandler handles Prometheus remote read requests.
-func RemoteReadHandler(q storage.Queryable, logger log.Logger) http.Handler {
+func RemoteReadHandler(q storage.SampleAndChunkQueryable, maxBytesInFrame int, lg log.Logger) http.Handler {
```

Comment:
We can remove `maxBytesInFrame` option and move the constant from handlers.go to this file and use it here.

Reply:
The reason why this variable is passed as an argument here is to allow testing the [scenario](https://github.com/grafana/mimir/blob/26eb01de7112e17de31d1189adad81c9197ca581/pkg/querier/remote_read_test.go#L187) in which a message has to be splitted into more than one frame due to [max frame size](https://github.com/grafana/mimir/blob/26eb01de7112e17de31d1189adad81c9197ca581/pkg/querier/remote_read_test.go#L245-L246) reached. 

The idea is to be able to do it without having push a whole megabyte series data (default max frame size) in a unit test and simplifying result assertion.

Alternatively, we may define this `maxBytesInFrame` constant as variable and overriding in the unit test, but based on my experience in previous PRs I think this is an discouraged practice.

Reply:
We can also define two functions, one used in tests with variable `maxBytesInFrame`, other one exported which calls the first function with hardcoded number.

Reply:
Definitely, that sounds like the best approach. :)

Reply:
Done @ 6019c81913ae717ec6f6e660fbf0713d3a7914ca